### PR TITLE
Created exterminatorGame, which allows users to spawn a pistol and shoot rats

### PR DESCRIPTION
**Pull request by ericrius1**
To enter the game: Run pistol.js and shoot at rats.
For every rat you kill, you get a point.
Your score will be displayed at https://desolate-bastion-1742.herokuapp.com/

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7ea0a626d9816ee241ae2e27e019b47fd5da139b available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1c8b3e394c3b35666bdfc859b0d35579b389189c available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bbfb282e37677ccf9de82c241016ff8fd6e61b17 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f1f601decbb20cbb7eb3e385aa51a04127c90764 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b79d8824e674221c77e67b0cc54c2ba310183e82 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 845a32b51d0f07c8cd588160d7b858034c0592c8 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: baf1a4dc4b41c74b7538df55638d788ec5f59494 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d4a8537e8b0dcc064a0960650121e0ccb69a3b73 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7e4fe4b521176908382029e52c103ccf89f389e1 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 005fd04585c2f59d0b2ba04d467656229089fe9d available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2a9bbe36126ed47afb020e9731724cb238e21369 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 90ef96746f64005120da24fb21c96cc6af4c8afc available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cb524d540d533e766c67512d2df294b8482fd597 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f256b1804cc102d27ab32fed138321e107721e4a available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eac91e71e84a9931ade7a6e938302d21e0987b65 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

File: examples/drylake/exterminatorGame/gameServer/app.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+var users = [];
+var connections = [];
+wsServer.on('request', function(request) {
+    console.log("SOMEONE JOINED");
+    var connection = request.accept(null, request.origin);
+    connections.push(connection);
+    connection.on('message', function(data) {
+        var userData = JSON.parse(data.utf8Data);
+        var user = _.find(users, function(user) {
```
Comment (imgntn) [inline examples/drylake/exterminatorGame/gameServer/app.js]: same for including underscore just to do this -- the built in array.filter method is pretty good for this kind of thing.  none of this is blocking re: interface , but i wouldn't merge it if we were making proper web apps :)

File: examples/drylake/exterminatorGame/gameServer/client/app.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+var React = require('react');
+var _ = require('underscore')
+var $ = require('jquery');
```
Comment (imgntn) [inline examples/drylake/exterminatorGame/gameServer/client/app.jsx]: normally its pretty heavy to include jquery just to use its ajax function.  not that load times are really our concern here, given that scripts will be less

File: examples/drylake/exterminatorGame/pistol.js [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+
+Script.include("../../libraries/utils.js");
+Script.include("../../libraries/constants.js");
+
+var GUN_FORCE = 10;
+
+
+var gameName = "Kill All The Rats!"
+// var HOST = "localhost:5000"
+var HOST = "desolate-bastion-1742.herokuapp.com";
```
Comment (imgntn) [inline examples/drylake/exterminatorGame/pistol.js]: i'm totally confused as to why you're using websockets for this and not a standard POST --or rather, why not either use express and build a rest api or use websockets and don't use express -- it's pretty redundant to have them both. were you just trying stuff out?  there's a lot going on re: react + websockets + express

Comment (imgntn) [inline examples/drylake/exterminatorGame/pistol.js]: this was great for the meeting but should probably be an environment variable or a placeholder before we merge

File: examples/drylake/exterminatorGame/rat.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+//
+//  Rat.js
+//  examples/toybox/entityScripts
```
Comment (imgntn) [inline examples/drylake/exterminatorGame/rat.js]: wrong header but we're probably getting rid of this..?

File: examples/drylake/exterminatorGame/ratSpawner.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+var scriptURL = Script.resolvePath('rat.js');
```
Comment (imgntn) [inline examples/drylake/exterminatorGame/ratSpawner.js]: i don't think we need this file anymore either

File: examples/particles.js [hunk 6]
```diff
@@ -1,93 +0,0 @@
-
```
Comment (imgntn) [inline examples/particles.js]: is this going to delete particles.js from master?  i'm not sure what happens with a merge, but we should make sure it doesnt get lost

File: examples/drylake/exterminatorGame/gameServer/client/app.jsx (hunk 2 above)
Comment (ericrius1) [inline examples/drylake/exterminatorGame/gameServer/client/app.jsx]: I'll remove jquery and underscore, and get rid of REST route before we merge this. First will work on refactoring gun

**Comment by hifi-gustavo** [comment]
Build console output for commit: fc9dfe8da5c2f956078e2e017c17a6d00e8f0ddf available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47c2119888b040b870e9fc471257800c9d481a0f available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4bf1c36fd6d759c8997461a4f3d5fc30e4843a8a available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d4c7b6b9b9dd69058585ddf0236bebdaee988c4a available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 53f75e305a5df54b286d85462fa2f46b0f9a791e available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8f0adb5337b487853e7ca931c5548d1c5cb2c253 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5f3da64a254abba9c3275c9368f6c94bdeee728e available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c5abd05c8d0ec4a77331ab6a7287a1a0b4195d9 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 688afa4c694b860b4287b245d163aba9598295ab available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 32f65eceaa779d398f9f6aefc3f92742826b1966 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 947d04e7f126714fab53e49f91bc5b3b2a3ab0dc available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8aa7228e5427752211f35c4ad6485d82ce264b0d available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 656d2f5c61ee10954f6d0dc85648d5d5c5ade12a available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0f4e531be6e8566c0790346c63543d4466c8409b available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c36719cba796ea005b754da6f09977b865902db0 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0980dbed6006448eaac5280fbbdef4baf27c77e8 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 224dea45e216dba25f8bf05c19b98e363dc777da available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3a9dcf56dffba84b307442e9a9f26fdda3a890cb available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c720c97e7096a5c01cf8128638dc4a0c61d66703 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 125c32b8fc010bec8b8131944a84a7f6004c163f available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by imgntn** [comment]
@ericrius1 what's the status of this PR?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22efcb1854ae45c2f937687f0db0ae370475ca98 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b7884f487c28b8737d0ad41c6f8d852e3d087f3e available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 706a67b9a1d5a2df6e0f081e79d985c82d06d452 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8992f6bb44e9e25da565ae5c1932b12e21d63d35 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b67c4817267819fd18e27029c3a259ceecbbf1e2 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 880aed0e6b494955fca2632832272a32fdd6eceb available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by RyanDowne** [comment]
It feels like we are waiting on this PR to just get a pistol in 'demo'. Can you make a separate one that just adds the pistol to the master script at the proper place on the table?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0e4aaaceeb2815bda3fe7f65afcebf7f6da92212 available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 77d66de7f377f4d46ffb7cfb4738f2c93c99fe9d available at: https://gist.github.com/f92c4ccdf955c25ff841<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6401.dmg

**Pull request merged**